src/components/conditional.js

```diff
@@ -78,6 +78,7 @@
     }, [checkCondition]);
 
     useEffect(() => {
+      if (!mounted.current) return;
```

Comment:
If an early return is done here the check on line 87 is then no longer necessary